### PR TITLE
Fix api documentation for System/System.Numerics

### PR DESCRIPTION
**Pull request by dakersnar**
Fixes https://github.com/dotnet/runtime/issues/72410, fixes part of https://github.com/dotnet/runtime/issues/72400. Also fixes a typo where APIs previously said "Tries to parses". 

Should mirror the source code changes here: https://github.com/dotnet/runtime/pull/76103.

### CODE REVIEW

File: xml/System.Numerics/Matrix3x2.xml
Comment (gewarren) [inline xml/System.Numerics/Matrix3x2.xml:1135]:
```suggestion
          <paramref name="column" /> was less than zero or greater than the number of columns.</exception>
```

File: xml/System.Numerics/Matrix4x4.xml
Comment (gewarren) [inline xml/System.Numerics/Matrix4x4.xml:2068]:
```suggestion
          <paramref name="column" /> was less than zero or greater than the number of columns.</exception>
```

**Review by gewarren — APPROVED** [review]

File: xml/System.Numerics/Matrix3x2.xml
Comment (carlossanlop) [inline xml/System.Numerics/Matrix3x2.xml:1135]:
```suggestion
          <para><paramref name="row" /> was less than zero or greater than the number of rows.</para>
          <para>-or-</para>
          <para><paramref name="column" /> was less than zero or greater than the number of columns.</para></exception>
```

File: xml/System.Numerics/Matrix4x4.xml
Comment (carlossanlop) [inline xml/System.Numerics/Matrix4x4.xml:2068]:
```suggestion
          <para><paramref name="row" /> was less than zero or greater than the number of rows.</para>
          <para>-or-</para>
          <para><paramref name="column" /> was less than zero or greater than the number of columns.</para></exception>
```

**Review by carlossanlop — APPROVED** [review]
All CI suggestions are a known issue in mdoc (operators don't respect inheritdoc).

Left the two suggestions that match your PR in runtime. After they get applied, it's good to merge.

**Comment by dakersnar** [comment]
Sorry for the double commits; github won't met me batch them.

**Comment by carlossanlop** [comment]
We forgive you, @dakersnar  😄 .

Did you attempt to batch them directly here in the Comments tab? Because you can't. You can only batch them in the Files tab.

**Comment by dakersnar** [comment]
@carlossanlop When I went to the files tab, only one of the two suggestions showed up there. This has happened twice today on this PR and I'm not sure what is causing it. Clicking "conversations" and selecting the hidden comment just gave me nothing. 

I will note that now I can see them, so I can't reproduce it.

**Comment by gewarren** [comment]
> @carlossanlop When I went to the files tab, only one of the two suggestions showed up there. This has happened twice today on this PR and I'm not sure what is causing it. Clicking "conversations" and selecting the hidden comment just gave me nothing.

Possibly you weren't looking at all commits?

**Comment by dakersnar** [comment]
That's possible. If it happens again, I'll try to dig into it more.

**Comment by carlossanlop** [comment]
The last 2 commits look good:

https://review.learn.microsoft.com/en-us/dotnet/api/system.numerics.matrix4x4.item?view=net-7.0&branch=pr-en-us-8423#exceptions
https://review.learn.microsoft.com/en-us/dotnet/api/system.numerics.matrix3x2.item?view=net-7.0&branch=pr-en-us-8423#exceptions

Since @gewarren already signed-off, and this was the only change left to verify, this is good to merge (and the runtime PR as well, assuming the changes there are identical to the ones here).

**Pull request merged**
